astor/code_gen.py

```diff
@@ -226,6 +226,32 @@ def body_or_else(self, node):
         self.body(node.body)
         self.else_body(node.orelse)
 
+    def docstring(self, node, indent=1):
+        """
+        Write the function/module/class docstring attribute
+
+        This method has an effect only for python >= 3.7 . For python earlier
+        versions the docstring was represented as the first expression in the
+        function/module/class body.
+
+        Args:
+            node (ast.AST): The node whose docstring is printed.
+            indent (int): How much to indent the docstring (usually is 0 for
+                ast.Module, and 1 for ast.FunctionDef, ast.AsyncFunctionDef and
+                ast.ClassDef).
+        """
+        if not sys.version_info >= (3, 7):
+            return
+
+        value = getattr(node, 'docstring', None)
+        if value is None:
+            return
+
+        self.newline()
+        self.indentation += indent
+        self.write(repr(value))
```

Comment:
I'm a bit unsure about using `repr(value)` for representing docstrings since this generates (unusual) single-quote strings:
```python
>>> import ast
>>> import astor
>>> srctxt = '''
def f():
    """docstring"""
    return 2
'''

>>> print(astor.to_source(ast.parse(srctxt)))
def f():
    'docstring'
    return 2
```

I thought about using the `astor.string_repr.pretty_string` function to generate triple-quoted docstrings,

https://github.com/berkerpeksag/astor/blob/b47718fa095e456c064d3d222f296fccfe36266b/astor/string_repr.py#L58

but I failed to understand its arguments
```python
def pretty_string(s, embedded, current_line, uni_lit=False,
                  min_trip_str=20, max_line=100):
```

What should I put as `embedded` and `current_line`?

Reply:
The meaning of `embedded` is documented at https://github.com/berkerpeksag/astor/blob/b47718fa095e456c064d3d222f296fccfe36266b/astor/code_gen.py#L537-L542 (perhaps we should move that snippet into a method or something?)

And see https://github.com/berkerpeksag/astor/blob/b47718fa095e456c064d3d222f296fccfe36266b/astor/code_gen.py#L549-L557 for the meaning of `current_line` (I don't really remember this one, though so I might be wrong.)